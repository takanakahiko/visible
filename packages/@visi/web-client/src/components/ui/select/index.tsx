export * from './select-mock';
export * from './select';
