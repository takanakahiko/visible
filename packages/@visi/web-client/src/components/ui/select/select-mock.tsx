import classNames from 'classnames';
import React, { useState } from 'react';

import { useTranslation } from '../../../utils/i18next';

export interface SelectMockProps {
  icon?: React.ReactNode;
  children: React.ReactNode;
  onOpen?(): void;
  onClose?(): void;
}

export const SelectMock = ({
  children,
  icon,
  onOpen,
  onClose,
}: SelectMockProps) => {
  const { t } = useTranslation();
  const [open, setOpen] = useState(false);

  const handleClick = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();

    if (open) {
      onClose?.();
    } else {
      onOpen?.();
    }

    setOpen(!open);
  };

  return (
    <details className="relative" open={open}>
      <summary
        className={classNames(
          'block',
          'text-sm',
          'border',
          'border-gray-400',
          'rounded',
          'text-gray-700',
          'py-1',
          'px-3',
          'w-24',
          'hover:bg-gray-100',
          'active:bg-gray-200',
          'transition-all',
          'duration-75',
          'ease-in',
          'text-left',
          'cursor-pointer',
          'truncate',
          'select-none',
        )}
        onClick={handleClick}
      >
        {icon && <span className="mr-1 text-xs text-gray-500">{icon}</span>}
        <span>{children}</span>
      </summary>

      <div className="absolute border border-gray-400 bg-white p-2 shadow z-10">
        {t('select.placeholder', 'This feature not implemented yet')}
      </div>
    </details>
  );
};
