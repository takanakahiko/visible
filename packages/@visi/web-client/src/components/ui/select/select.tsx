import { faChevronDown } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classNames from 'classnames';
import React from 'react';

export type SelectProps = JSX.IntrinsicElements['select'] & {
  children: React.ReactNode;
  className?: string;
  onOpen?(): void;
  onClose?(): void;
};

export const Select = ({ children, className, ...rest }: SelectProps) => {
  return (
    <div className={classNames('relative', 'text-sm', 'text-gray-700')}>
      <select
        className={classNames(
          'appearance-none',
          'py-1',
          'px-2',
          'border',
          'rounded',
          'text-left',
          'border-gray-400',
          'cursor-pointer',
          className,
        )}
        {...rest}
      >
        {children}
      </select>

      <FontAwesomeIcon
        className="absolute top-0 right-0 bottom-0 m-auto mr-2"
        icon={faChevronDown}
      />
    </div>
  );
};

export type OptionProps = JSX.IntrinsicElements['option'];

Select.Option = (props: OptionProps) => {
  return <option {...props} />;
};
