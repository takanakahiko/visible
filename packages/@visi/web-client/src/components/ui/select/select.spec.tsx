import { render } from '@testing-library/react';
import React from 'react';

import { Normal } from './select.stories';

describe('Select', () => {
  it('matches snapshot', () => {
    const { container } = render(<Normal />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
