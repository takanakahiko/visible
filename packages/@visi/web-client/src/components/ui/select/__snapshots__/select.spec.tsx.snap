// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select matches snapshot 1`] = `
<div
  class="relative text-sm text-gray-700"
>
  <select
    class="appearance-none py-1 px-2 border rounded w-64 text-left border-gray-400 cursor-pointer"
  >
    foo
  </select>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-chevron-down fa-w-14 absolute top-0 right-0 bottom-0 m-auto mr-2"
    data-icon="chevron-down"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
      fill="currentColor"
    />
  </svg>
</div>
`;
